general-concepts/slotting/text.xml

```diff
@@ -45,6 +45,102 @@ To <c>DEPEND</c> upon a package in a specific slot, refer to
 </p>
 </body>
 
+<section>
+<title>Considerations for slotting</title>
+<body>
+
+<p>
+Slotting is a powerful tool and as such, is easily abused. It's critical
+that when slotting, package maintainers carefully consider the impact
+on <e>reverse dependencies</e>.
+</p>
+
+<p>
+Do not be fooled by the ease in adapting the package itself to slotting:
+slotting the package itself is always the easiest part.
+</p>
+
+<p>
+Before proceeding at all, consider the impact on the wider ecosystem
+beyond Gentoo. Especially examine other distributions and their handling
+of the problem in question. Gentoo cannot unilaterally move software to
+a non-standard location.
+</p>
+
+<p>
+Even if every possible consumer of a slotted package has been adapted in
+Gentoo, remember that people may develop or test software on Gentoo systems:
+they both want to be able to build that software without having to adapt
+to Gentooisms, and they also don't want to unintentionally write their
+software such that it relies on Gentooisms.
+</p>
+
+<p>
+Ecosystems that lend themselves to slotting, such as Guile or Lua, cannot
+be naively slotted by just adapting the package itself. Other packages
+must be primed to handle a slotted world: an eclass mechanism usually needs
+to exist for packages to declare which versions they support, as well as
+force building against the version currently depended on to avoid automaigc
+dependencies.
+</p>
+
+<p>
+All reverse dependencies of a package should be tested and examined when
+considering any slotting approach, in particular asking the following:
+</p>
+
+<ul>
+  <li>how does this software use this package;</li>
+  <li>how the build system should locate the (right) version of this package;</li>
+  <li>how the package (and the right version of it) should be located at runtime,
+      if required.</li>
+</ul>
+
+</body>
+
+<subsection>
+<title>Libraries</title>
+<body>
+
+<p>
+As a rule of thumb, libraries should not be slotted unless upstream
+explicitly intend (major) versions to be installable in parallel. The
+presumption should be that a library cannot be slotted until proven otherwise.
+</p>
+
```

Comment:
Is `dev-java/bcprov` regarded a `library`? If so, what' should have been the alternative to slotting in commit [901da76177](https://gitweb.gentoo.org/repo/gentoo.git/commit/dev-java/bcprov/bcprov-1.74.ebuild?id=901da761770cc5e16a9303ddcb0722f5b3654773) followed-up by commit [154e9224cf](https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=154e9224cf297219ed0a5baf3cbc1e5eb5c42456), all was done according to [this wiki article](https://wiki.gentoo.org/wiki/Java_Developer_Guide#Slots)?

Reply:
I'd say Java libraries are by construction intended to be installable in parallel (they are often not mean to be consumed in parallel, but that is unrelated to slotting).